* Don't start a timed login if an username was already entered.
* Debian #157792, #160578, #184218; handle PAM's AUTHINFO_UNAVAIL in incorrect
  password error handler.

XXX To be splitted further and clarified

Index: gdm-2.18.1/daemon/verify-pam.c
===================================================================
--- gdm-2.18.1.orig/daemon/verify-pam.c	2007-04-09 07:15:15.000000000 +0200
+++ gdm-2.18.1/daemon/verify-pam.c	2007-05-02 10:38:29.000000000 +0200
@@ -826,17 +826,18 @@
     started_timer = FALSE;
     null_tok = 0;
 
-    /* start the timer for timed logins */
-    if ( ! ve_string_empty (gdm_get_value_string (GDM_KEY_TIMED_LOGIN)) &&
-	d->timed_login_ok &&
-	(local || gdm_get_value_bool (GDM_KEY_ALLOW_REMOTE_AUTOLOGIN))) {
-	    gdm_slave_greeter_ctl_no_ret (GDM_STARTTIMER, "");
-	    started_timer = TRUE;
-    }
-
+    /* don't start a time login if we've already entered a username */
     if (username != NULL) {
 	    login = g_strdup (username);
 	    gdm_slave_greeter_ctl_no_ret (GDM_SETLOGIN, login);
+    } else {
+	    /* start the timer for timed logins */
+	    if ( ! ve_string_empty (gdm_get_value_string (GDM_KEY_TIMED_LOGIN)) &&
+		d->timed_login_ok &&
+		(local || gdm_get_value_bool (GDM_KEY_ALLOW_REMOTE_AUTOLOGIN))) {
+		    gdm_slave_greeter_ctl_no_ret (GDM_STARTTIMER, "");
+		    started_timer = TRUE;
+		}
     }
 
     cur_gdm_disp = d;
@@ -1165,8 +1166,10 @@
     if ( ! error_msg_given &&
 	gdm_slave_action_pending ()) {
 	    /* I'm not sure yet if I should display this message for any other issues - heeten */
+	    /* at least AUTHINFO_UNAVAIL -- it's what user unknown is! */
 	    if (pamerr == PAM_AUTH_ERR ||
-		pamerr == PAM_USER_UNKNOWN) {
+		pamerr == PAM_USER_UNKNOWN ||
+		pamerr == PAM_AUTHINFO_UNAVAIL) {
 		    gboolean is_capslock = FALSE;
 		    const char *basemsg;
 		    char *msg;
